src/net/sourceforge/kolmafia/textui/parsetree/Function.java

```diff
@@ -179,38 +186,43 @@ private boolean paramsMatchNoVararg(final List<? extends TypedNode> params, Matc
       switch (match) {
         case EXACT:
           if (!currentParam.getRawType().equals(currentValue.getRawType())) {
-            matched = false;
+            return false;
           }
           break;
 
         case BASE:
           if (!paramType.equals(valueType)) {
-            matched = false;
+            return false;
           }
           break;
 
         case COERCE:
           if (!Operator.validCoercion(paramType, valueType, "parameter")) {
-            matched = false;
+            return false;
           }
           break;
       }
     }
 
-    if (matched && !refIterator.hasNext() && !valIterator.hasNext()) {
-      return true;
+    if (refIterator.hasNext() || valIterator.hasNext()) {
+      return false;
     }
-    return false;
+
+    return true;
   }
 
   private boolean paramsMatchVararg(final List<? extends TypedNode> params, MatchType match) {
+    // on a vararg match, the last parameter can match 0 values so wee need at least X-1 values
```

Comment:
```suggestion
    // on a vararg match, the last parameter can match 0 values so we need at least X-1 values
```
trivial spelling issue

Reply:
Just a wee typo. Will fix it in no time.